### DISCUSSION TITLE
Help wanted to write a SWC plugin version of babel for nextjs, also for turbopack

### DISCUSSION THREAD

**Discussion by Jonatthu** [nativewind/nativewind#306] [General]
As the title says so, I want any guidelines or help to get started into making a plugin in rust to make sure nativewind can run on nextjs using SWC and also the new turbopack for nextjs13

Any advice to get started quickly?

https://github.com/swc-project/plugins/issues/103

**Comment by marklawlor** [comment]
The babel plugin is optional and NativeWind works on Next.js with SWC if you used the `styled()` wrapper.

I'm not an expert of Rust and I've never written an SWC plugin before so I unfortunately don't have any advise to provide.

I'm going to move this is to a discussion in case in case a community member can assist

**Reply by esinanturan** [reply]
@marklawlor How about a vite plugin ? How to make styles work without need of styled() wrapper in vite. Its done by babel plugin but I could't make it work with vite. It would be pretty useful considering making nativewind support vite. Everything works fine except styles are not applied when using components without styled HOC.

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
